components/PeerReviews/PeerReviewRequestCard.tsx

```diff
@@ -0,0 +1,156 @@
+import { ReactElement, useState } from "react";
+import { PeerReviewRequest } from "~/config/types/peerReview";
+import { css, StyleSheet } from "aphrodite";
+import ALink from "~/components/ALink";
+import AuthorAvatar from "~/components/AuthorAvatar";
+import colors from "~/config/themes/colors";
+import { getUrlToUniDoc } from "~/config/utils/routing";
+import { timeSince } from "~/config/utils/dates";
+import DropdownButton from "~/components/Form/DropdownButton";
+import { ID } from "~/config/types/root_types";
+import PeerReviewRequestModal from "./PeerReviewInviteModal";
+import PeerReviewPerson from "./PeerReviewPerson";
+
+
+type Props = {
+  peerReviewRequest: PeerReviewRequest;
+  fetchReviewRequests: Function;
+};
+
+function PeerReviewRequestCard({
+  peerReviewRequest,
+  fetchReviewRequests,
+}: Props): ReactElement {
+
+  const opts = [{
+    label: "Invite",
+    value: "invite"
+  }, {
+    label: "Decline",
+    value: "decline"
+  }, {
+    label: "Close",
+    value: "close"    
+  }];
+
+  const [manageDropdownOpenFor, setManageDropdownOpenFor] = useState<ID>(null)
+  const [isInviteModalOpen, setsIsInviteModalOpen] = useState<Boolean>(false);
+  const author = peerReviewRequest?.requestedByUser?.authorProfile
+  const invites = peerReviewRequest?.invites ?? [];
+
+  return (
+    <div className={css(styles.PeerReviewRequestCard)}>
+      <PeerReviewRequestModal
+        peerReviewRequest={peerReviewRequest}
+        isOpen={isInviteModalOpen}
+        closeModal={() => setsIsInviteModalOpen(false)}
+        fetchReviewRequests={fetchReviewRequests}
+      />
+      <div className={css(styles.detailsContainer)}>
+        <span className={css(styles.avatarContainer)}>
+          <AuthorAvatar
+            author={(peerReviewRequest?.requestedByUser ?? {})["authorProfile"]}
+            fontSize={16}
+            size={35}
+            spacing={5}
+          />
+        </span>
+        <div>
+          <div className={css(styles.mainTextRow)}>
+            <ALink href={getUrlToUniDoc(peerReviewRequest?.unifiedDocument)}>
+              {author?.firstName} {author?.lastName}
+            </ALink>
+            <span> has requested a peer review on </span>
+            <ALink href={getUrlToUniDoc(peerReviewRequest?.unifiedDocument)}>
+              {peerReviewRequest?.unifiedDocument?.document?.title}
+            </ALink>
+          </div>
+          <div className={css(styles.dateRow)}>
+            {timeSince(peerReviewRequest.createdDate)}
+          </div>
+          {invites.length > 0 &&
+            <div className={css(styles.invitedRow)}>
+              {invites.map((invite) => (
+                <PeerReviewPerson
+                  id={invite?.id}
+                  status={invite?.status}
+                  authorProfile={invite?.recipient?.authorProfile}
+                />
+              ))}
```

Comment:
Mapping components within `render` is usually not encouraged because of (again) memoization / lifecycle reasons. It's usually better to assign this to a variable outside of `render` than insert as `{yourVariable}` within render

Reply:
Makes sense